openwpm/deploy_browsers/selenium_firefox.py

```diff
@@ -129,7 +122,7 @@ def __init__(
         self.service_args = service_args or []
 
     def command_line_args(self):
-        return ["--port", "%d" % self.port]
+        return ["--port", "%d" % self.port] + self.service_args
 
     def send_remote_shutdown_command(self):
```

Comment:
Do you know why we pass here? What are we trying to supress/stop from happening?

Reply:
Do you mean why we pass in `send_remote_shutdown_command()`? class `PatchedGeckoDriverService` is a patched version of [Selenium's `Service` class](https://github.com/SeleniumHQ/selenium/blob/0849604f66a3e15e7ae650cfb0ac870328ce271f/py/selenium/webdriver/firefox/service.py) and we've copied this method from there.

Reply:
Ah okay, thank you. 
Is there a reason we shouldn't inherit from the Firefox service and overwrite `__init__` instead of having to keep up with the changes in Selenium?

Reply:
I was wrong. `PatchedGeckoDriverService` is a subclass of the `Service` class in [`selenium.webdriver.common.service`](https://github.com/SeleniumHQ/selenium/blob/0849604f66a3e15e7ae650cfb0ac870328ce271f/py/selenium/webdriver/common/service.py) and not [`selenium.webdriver.firefox.service`](https://github.com/SeleniumHQ/selenium/blob/0849604f66a3e15e7ae650cfb0ac870328ce271f/py/selenium/webdriver/firefox/service.py) as I initially thought. That explains why it has to define the`command_line_args()` and `send_remote_shutdown_command()` methods.

I guess we could still do what you are suggesting though. Do you think we should?

Reply:
Yes, because this way we only need to check what changed in the `__init__` Method of `selenium.webdriver.firefox.service` and not all changes to that class.